ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -1021,6 +1042,39 @@ - (ORKStepViewController *)viewControllerForStep:(ORKStep *)step {
         
         stepViewController.restorationIdentifier = step.identifier;
         stepViewController.restorationClass = stepViewControllerClass;
+        if ([stepViewController isKindOfClass:[ORKReviewStepViewController class]]) {
+            ORKReviewStepViewController *reviewStepViewController = (ORKReviewStepViewController *) stepViewController;
+            NSMutableArray *steps = [[NSMutableArray alloc] init];
+            if (reviewStepViewController.reviewStep.isStandalone) {
+                for (ORKStep *step in reviewStepViewController.reviewStep.steps) {
+                    if ([self shouldReviewStep:step]) {
+                        [steps addObject:step];
+                    }
+                }
+            } else {
+                ORKStep *nextStep = step;
+                do {
+                    switch (reviewStepViewController.reviewStep.reviewDirection) {
```

Comment:
That has 2 major downsides: (1) Since `_managedStepIdentifiers` only contains the identifiers, we need to call `stepWithIdentifier:` to get the actual step. That will only work if `stepWithIdentifier:` is implemented. (2) If you want to start a task with a review step, `managedStepIdentifiers` only contains the review step but none of the other steps that should be reviewed. I would keep the current implementation instead.